src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -223,24 +223,49 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
   /* Public for testing. */
   public static void addCourse(HttpServletRequest request, DatastoreService db) {
-    AddSchoolData addSchool = new AddSchoolData();
-    addSchool.addSchoolData(db, request);
-
     String name = request.getParameter("course-name");
     String prof = request.getParameter("prof-name");
     Long units = Long.parseLong(request.getParameter("num-units"));
     String term = request.getParameter("term");
     String school = request.getParameter("school-name");
 
-    Entity newCourse = new Entity("Course-Info");
-    newCourse.setProperty("name", name);
-    newCourse.setProperty("search-name", name.toLowerCase());
-    newCourse.setProperty("professor", prof);
-    newCourse.setProperty("search-professor", prof.toLowerCase());
-    newCourse.setProperty("units", units);
-    newCourse.setProperty("term", term);
-    newCourse.setProperty("school", school); // Already lowercase.
-    db.put(newCourse);
+    Boolean isNewCourse = checkNewCourse(name, prof, units, term, school, db);
+
+    if (isNewCourse) {
+      Entity newCourse = new Entity("Course-Info");
+      newCourse.setProperty("name", name);
+      newCourse.setProperty("search-name", name.toLowerCase());
+      newCourse.setProperty("professor", prof);
+      newCourse.setProperty("search-professor", prof.toLowerCase());
+      newCourse.setProperty("units", units);
+      newCourse.setProperty("term", term);
+      newCourse.setProperty("school", school); // Already lowercase.
+      db.put(newCourse);
+
+      AddSchoolData addSchool = new AddSchoolData();
+      addSchool.addSchoolData(db, request);
+    }
+  }
+
+  private static Boolean checkNewCourse(
+      String name, String prof, Long units, String term, String school, DatastoreService db) {
+    List<Filter> filters = new ArrayList<>();
+    filters.add(new FilterPredicate("search-name", FilterOperator.EQUAL, name.toLowerCase()));
+    filters.add(new FilterPredicate("search-professor", FilterOperator.EQUAL, prof.toLowerCase()));
+    filters.add(new FilterPredicate("units", FilterOperator.EQUAL, units));
+    filters.add(new FilterPredicate("term", FilterOperator.EQUAL, term));
+    filters.add(new FilterPredicate("school", FilterOperator.EQUAL, school));
+
+    Query courseQuery = new Query("Course-Info");
+    if (!filters.isEmpty()) {
+      if (filters.size() == 1) {
```

Comment:
When would filters be empty or size of only 1? Aren't you adding 5 filters directly above here?

Reply:
good point, I think that was just a reflex stuck from the search in doGet